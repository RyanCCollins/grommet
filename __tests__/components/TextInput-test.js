// (C) Copyright 2014-2016 Hewlett Packard Enterprise Development LP

import React from 'react';
import renderer from 'react/lib/ReactTestRenderer';

import TextInput from '../../src/js/components/TextInput';

// needed because this:
// https://github.com/facebook/jest/issues/1353
jest.mock('react-dom');

describe('TextInput', () => {
  it('has correct default options', () => {
    const component = renderer.create(
       <TextInput id="item1" name="item-1" value="one" />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
