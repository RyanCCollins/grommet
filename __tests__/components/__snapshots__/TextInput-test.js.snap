exports[`TextInput has correct default options 1`] = `
<input
  autoComplete="off"
  className="grommetux-text-input grommetux-input"
  defaultValue={undefined}
  id="item1"
  name="item-1"
  onChange={[Function bound _onInputChange]}
  onFocus={[Function bound _onFocus]}
  placeholder={undefined}
  value="one" />
`;
